lib/chef-vault/item_keys.rb

```diff
@@ -28,10 +28,52 @@ def initialize(vault, name)
       @raw_data["admins"] = []
       @raw_data["clients"] = []
       @raw_data["search_query"] = []
+      @raw_data["mode"] = "default"
+      @cache = {} # write-back cache for keys
+    end
+
+    def [](key)
+      # return options immediately
+      return @raw_data[key] if %w{id admins clients search_query mode}.include?(key)
+      # check if the key is in the write-back cache
+      ckey = @cache[key]
+      return ckey unless ckey.nil?
+      # check if the key is saved in sparse mode
+      spath = "#{@raw_data["id"]}_key_#{key}"
+      skey = if Chef::Config[:solo_legacy_mode]
+               load_solo(spath)
+             else
+               begin
+                 Chef::DataBagItem.load(@data_bag, spath)
+               rescue Net::HTTPServerException => http_error
+                 nil if http_error.response.code == "404"
+               end
+             end
+      if skey
+        skey[key]
+      else
+        # fallback to raw data
+        @raw_data[key]
+      end
     end
 
     def include?(key)
-      @raw_data.keys.include?(key)
+      # check if the key is in the write-back cache
+      ckey = @cache[key]
+      return (ckey ? true : false) unless ckey.nil?
```

Comment:
OPT:
(if you really want to return true value)
```
return !!ckey if ckey
```

or if you just want to return true-ish value
```
return ckey if ckey
```